utils/src/main/java/uk/gov/pay/commons/utils/xray/XRaySessionProfiler.java

```diff
@@ -36,37 +40,23 @@ public void endOperationProfile(String operationName, DatabaseQuery databaseQuer
 
     @Override
     public Object profileExecutionOfQuery(DatabaseQuery databaseQuery, Record record, AbstractSession abstractSession) {
-
-        HashMap additionalParams = new HashMap();
-        DatabaseMetaData metadata;
-        String hostname = "database";
-        try {
-            Connection connection = abstractSession.getAccessor().getConnection();
-            metadata = connection.getMetaData();
-            additionalParams.put("url", metadata.getURL());
-            additionalParams.put("user", metadata.getUserName());
-            additionalParams.put("driver_version", metadata.getDriverVersion());
-            additionalParams.put("database_type", metadata.getDatabaseProductName());
-            additionalParams.put("database_version", metadata.getDatabaseProductVersion());
-            additionalParams.put("preparation", databaseQuery.isCallQuery() ? "call" : "statement");
-            additionalParams.put("sanitized_query", StringUtils.isEmpty(databaseQuery.getSQLString()) ? "" : databaseQuery.getSQLString());
-
-            try {
-                hostname = new URI((new URI(metadata.getURL())).getSchemeSpecificPart()).getHost();
-                hostname = connection.getCatalog() + "@" + hostname;
-            } catch (URISyntaxException exception) {
-                logger.warn("Error parsing database host name.");
+        if (recorder.getCurrentSegmentOptional().isPresent()) {
+            final Optional<Connection> maybeConnection = Optional.ofNullable(abstractSession.getAccessor()).map(Accessor::getConnection);
+            Map<String, Object> databaseMetadata = new HashMap<>();
+            String hostname = "database";
+            if (maybeConnection.isPresent()) {
+                final Connection connection = maybeConnection.get();
+                databaseMetadata = getMetadata(databaseQuery, connection);
+                hostname = getDatabaseHostName(connection);
+            } else {
+                logger.warn("There is no database connection available");
```

Comment:
Should this (and possibly other logging in this class) be `error`? Something’s gone pretty wrong if we don’t have a database connection!